src/main/java/fr/insalyon/creatis/gasw/plugin/listener/healing/execution/CommandState.java

```diff
@@ -308,7 +320,103 @@ public void addUploadTime(long time) {
         this.outputTimes.add(time);
     }
 
-    /**
+    private void computeJobErrorRate() throws DAOException {
+        JobDAO jobDAO = DAOFactory.getDAOFactory().getJobDAO();
+        this.jobErrorRate = 100.0 * jobDAO.getFailedByCommand(this.command).size() / jobDAO.getJobsByCommand(this.command).size();
+        logger.info("[Healing] Updated the jobErrorRate to " + this.jobErrorRate);
+
+    }
+
+    private void computeInvocationPartialErrorRate() throws DAOException {
+        // TODO : after further analysis, also consider jobs running for more than MAX hours when computing invocationPartialErrorRate
+        int failures = 0;
+        JobDAO jobDAO = DAOFactory.getDAOFactory().getJobDAO();
+        List<Integer> invocationIDs = jobDAO.getInvocationsByCommand(this.command);
+        for (int invocation : invocationIDs) {
+            if (jobDAO.getFailedJobsByInvocationID(invocation).size() >= 1) {
+                failures++;
+            }
+        }
+        this.invocationPartialErrorRate = 100.0 * failures / jobDAO.getInvocationsByCommand(this.command).size();
```

Comment:
could use `invocationIDs.size()` to not repeat the database request.

Reply:
Indeed